include/exec/at_coroutine_exit.hpp

```diff
@@ -0,0 +1,256 @@
+    }
+
+  template <class _Value> _Value&& operator()(_Value&& __value) const noexcept {
+    return (_Value &&) __value;
+  }
+};
+inline constexpr __die_on_stop_t __die_on_stop;
+
+template <class... _Ts>
+  class [[nodiscard]] __task {
+    struct __promise;
+  public:
+    using promise_type = __promise;
+
+    explicit __task(__coro::coroutine_handle<__promise> __coro) noexcept
+    : __coro_(__coro) {}
+
+    __task(__task&& __that) noexcept
+    : __coro_(std::exchange(__that.__coro_, {})) {}
+
+    bool await_ready() const noexcept { return false; }
+
+    template <typename _Promise>
+        requires requires (_Promise& __promise, __coro::coroutine_handle<promise_type> __h) {
+          { __promise.continuation() } -> std::convertible_to<__coro::coroutine_handle<>>;
+          { __promise.set_continuation(__h) };
+        }
+      bool await_suspend(__coro::coroutine_handle<_Promise> __parent) noexcept {
+        __coro_.promise().__coro_ = __parent.promise().continuation();
+        __parent.promise().set_continuation(__coro_);
+        return false;
+      }
+
+    std::tuple<_Ts&...> await_resume() noexcept {
+      return std::exchange(__coro_, {}).promise().__args_;
+    }
+
+  private:
+    struct __final_awaitable {
+      static std::false_type await_ready() noexcept {
+        return {};
+      }
+
+      static __coro::coroutine_handle<>
+      await_suspend(__coro::coroutine_handle<__promise> __h) noexcept {
+        auto __coro = __h.promise().__coro_;
+        __h.destroy();
+        return __coro;
+      }
+
+      void await_resume() const noexcept {
+      }
+    };
+
+    struct __promise : stdexec::with_awaitable_senders<__promise> {
+      template <typename _Action>
+        explicit __promise(_Action&&, _Ts&... __ts) noexcept
+        : __args_{__ts...} {
+        }
+
+      __coro::suspend_always initial_suspend() noexcept {
+        return {};
+      }
+      __final_awaitable final_suspend() noexcept {
+        return {};
+      }
+
+      void return_void() noexcept {
+      }
+
+      [[noreturn]] void unhandled_exception() noexcept {
+        std::terminate();
+      }
+
+      __coro::coroutine_handle<__promise> unhandled_stopped() noexcept {
+        return __coro::coroutine_handle<__promise>::from_promise(*this);
+      }
+
+      __task get_return_object() noexcept {
+        return __task(__coro::coroutine_handle<__promise>::from_promise(*this));
+      }
+
+      template <class _Awaitable>
+        decltype(auto) await_transform(_Awaitable&& __awaitable) noexcept {
+          return stdexec::as_awaitable(__die_on_stop((_Awaitable &&) __awaitable), *this);
+        }
+
+      __coro::coroutine_handle<> __coro_{};
+      std::tuple<_Ts&...> __args_{};
+    };
+
+    __coro::coroutine_handle<__promise> __coro_;
+  };
+
+struct __at_coroutine_exit_t {
+private:
+  template <typename _Action, typename... _Ts>
+    static __task<_Ts...> at_coroutine_exit(_Action&& __action, _Ts&&... __ts) {
```

Comment:
```suggestion
    static __task<_Ts...> at_coroutine_exit(_Action __action, _Ts... __ts) {
```

Reply:
This fixes some nasty compile errors when looking for a promise constructor overload. But I don't quite understand why yet...